### ISSUE TITLE
Extend ObsoleteAttribute

### ISSUE THREAD

**Issue by terrajobst** [dotnet/runtime#33089]
We'd like to [make obsoletion more viable](https://github.com/dotnet/designs/blob/master/accepted/2020/better-obsoletion/better-obsoletion.md). This requires extending the `ObsoleteAttribute`:

### API

```C#
#nullable enable
namespace System
{
    public partial class ObsoleteAttribute : Attribute
    {
        // Existing:
        //
        // public ObsoleteAttribute();
        // public ObsoleteAttribute(string message);
        // public ObsoleteAttribute(string message, bool error);
        // public bool IsError { get; }
        // public string Message { get; }

        // New:
        public string? DiagnosticId { get; set; }
        public string? UrlFormat { get; set; }
    }
}
```

### Usage

```C#
namespace System.Collections
{
    [Obsolete(DiagnosticId="BCL0006", UrlFormat="https://aka.ms/obsolete/{0}")]
    public class ArrayList
    {
        // ...
    }

    [Obsolete(DiagnosticId="BCL0006", UrlFormat="https://aka.ms/obsolete/{0}")]
    public class Stack
    {
        // ...
    }
}
namespace System.Security
{
    [Obsolete(DiagnosticId"BCL0001", UrlFormat="https://aka.ms/obsolete/{0}")]
    public sealed class SecureString
    {
        // ...
    }
}
```

**Comment by terrajobst** [comment]
Resolved by #33248

**Comment by GrabYourPitchforks** [comment]
BTW I'm experimenting with using the _UrlFormat_ property for new obsolete messages, and it's running afoul of CA2243.

```txt
Severity	Code	Description	Project	File	Line	Suppression State
Error	CA2243	In the constructor of 'ObsoleteAttribute', change the value of argument 'ObsoleteAttribute.UrlFormat', which is currently "https://aka.ms/dotnet-obsoletions/fill-me-in/{0}", to something that can be correctly parsed as 'Uri'.	System.Runtime.Serialization.Formatters (net5.0-Unix), System.Runtime.Serialization.Formatters (net5.0-Windows_NT)	C:\runtime.titan\src\libraries\System.Runtime.Serialization.Formatters\src\System\Runtime\Serialization\Formatters\Binary\BinaryFormatter.cs	41	Active
```

We should update CA2243 to be aware of this particular property and to ignore it, or we should suppress CA2243 globally throughout the runtime repo, or we should rename the property to something other than _UrlFormat_.

Any preference as to these courses of action?
